### PR TITLE
[ENH] warn when we can't implant mxc db, or if storage is almost full.

### PR DESCRIPTION
**Pull request by rksh**
- seeing what's left of our SQL problems if we tell people not to stumble with full devices.
- language pack updates... the endless struggle.

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java [hunk 1]
```diff
@@ -356,8 +360,24 @@ public void onCreate(Bundle savedInstanceState) {
         //if (!state.mxcDbHelper.isPresent()) {
         try {
             state.mxcDbHelper.implantMxcDatabase();
+        } catch (InsufficientSpaceException isex) {
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java]: hot

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java [hunk 2]
```diff
@@ -2499,4 +2519,29 @@ public void backgroundUploadFile(){
             MainActivity.warn("Authentication failure on background run upload");
         }
     }
+
+    public boolean checkStorage() {
+        boolean safe;
+        boolean external = hasSD();
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java]: Is this only called after the initial setup where we ask for permission for the external storage? Want to make sure it's happy on initial install.

Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java]: oh good point.

Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java]: looks like `onCreate` calls `setupPermissions` almost immediately - you might still get the warning if you haven't agreed and local storage is really tight, but on "accept", you'd relaunch with the new perms, perform the check again, and get `hasSD() == true`

**Pull request merged**